content/en/docs/tasks/security/authorization/authz-td-migration/index.md

```diff
@@ -0,0 +1,202 @@
+
+    {{< text bash >}}
+    $ kubectl delete pod --all -n sleep-allow
+    {{< /text >}}
+
+1. Verify that requests to `httpbin` from both `sleep` in `default` namespace and `sleep-allow` namespace are denied.
+
+    {{< text bash >}}
+    $ kubectl exec $(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name}) -c sleep -- curl http://httpbin.default:8000/ip -s -o /dev/null -w "%{http_code}\n"
+    403
+    {{< /text >}}
+
+    {{< text bash >}}
+    $ kubectl exec $(kubectl -n sleep-allow get pod -l app=sleep -o jsonpath={.items..metadata.name}) -c sleep -n sleep-allow -- curl http://httpbin.default:8000/ip -s -o /dev/null -w "%{http_code}\n"
+    403
+    {{< /text >}}
+
+    This is because we specified an authorization policy that deny all requests to `httpbin`, except the ones
+     the `old-td/ns/sleep-allow/sa/sleep` identity, which is the old identity of the `sleep` application in `sleep-allow` namespace.
+     When we migrated to a new trust domain above, i.e. `new-td`, the identity of this `sleep` application is now `new-td/ns/sleep-allow/sa/sleep`,
+     which is not the same as `old-td/ns/sleep-allow/sa/sleep`. Therefore, requests from the `sleep` application in `sleep-allow` namespace
+     to `httpbin` were allowed before are now being denied. Prior to Istio 1.4, the only way to make this work is to change the authorization
+     policy manually. In Istio 1.4, we introduce an easy way, as shown below.
+
+## Migrate trust domain with trust domain aliases
+
+1. Prepare trust domain aliases for the Istio mesh.
+
+    {{< text bash >}}
+    $ echo "apiVersion: install.istio.io/v1alpha2
+      kind: IstioControlPlane
+      spec:
+        values:
+          global:
+            mtls:
+              enabled: true
+            trustDomain: new-td
+            trustDomainAliases:
+              - old-td" > td-installation.yaml
+    {{< /text >}}
+
+1. Repeat step 2-4 from [Migrate trust domain without trust domain aliases](/docs/tasks/security/authorization/authz-td-migration/#migrate-trust-domain-without-trust-domain-aliases).
+   Istio mesh is now running with the new trust domain `new-td` as before. However, it recognizes that `old-td` used to be an equivalent trust domain. Therefore,
+   it will treat `old-td` the same as `new-td`.
+1. Without changing the authorization policy, verify that requests to `httpbin` from:
+
+    * `sleep` in the `default` namespace are denied.
+
+    {{< text bash >}}
+    $ kubectl exec $(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name}) -c sleep -- curl http://httpbin.default:8000/ip -s -o /dev/null -w "%{http_code}\n"
+    403
+    {{< /text >}}
+
+    * `sleep` in the `sleep-allow` namespace are allowed.
+
+    {{< text bash >}}
+    $ kubectl exec $(kubectl -n sleep-allow get pod -l app=sleep -o jsonpath={.items..metadata.name}) -c sleep -n sleep-allow -- curl http://httpbin.default:8000/ip -s -o /dev/null -w "%{http_code}\n"
+    200
+    {{< /text >}}
+
+## Best practices
```

Comment:
@liminw Actually, should we recommend using "cluster.local' as a pointer? I think this could be bad if we want to distinguish between trust domains in a multi cluster environment. Plus, the only benefit of using "cluster.local" is so that the users do not have to include the aliases, which I don't think is a big improvement. Wdyt?

Reply:
I think the benefit is two folds. 

* For users, if they always use "cluster.local" in authorization policy. When they migrate from one trust domain to the other, they do not need to think of setting aliases. It will simply work. 

* For us (developer and feature maintainer of authz feature), if all users use "cluster.local" in authorization policy to refer to local trust domain, we do not need to generate duplicate policies for each trust domain aliases, which will simply the code and improve performance.